core/block_events.js

```diff
@@ -56,6 +56,7 @@ Blockly.Events.BlockBase = function(block) {
    */
   this.blockId = block.id;
   this.workspaceId = block.workspace.id;
+  this.workspace = block.workspace;
```

Comment:
Why did you add this?  Both places where you want the workspace already have a local variable representing the workspace, and at any rate only the workspace ID is serialized.